### PR TITLE
Update framework version and support more tests

### PR DESCRIPTION
**Pull request by jamming**
Update framework version and support more tests (with different backends and different databases).

### CODE REVIEW

File: frameworks/Java/sabina/jetty-mongodb.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+./setup.sh -Dsabina.backend=jetty -Dsabina.benchmark.repository=mongodb
```
Comment (hamiltont) [inline frameworks/Java/sabina/jetty-mongodb.sh]: ping @msmith-techempower - FYI, you'll notice that the `&` is no longer required with the new way we call the setup files. Cool, huh? :-)

File: frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        final String DATABASE = settings.getProperty ("mongodb.database");
+        final String WORLD = settings.getProperty ("mongodb.world.collection");
+        final String FORTUNE = settings.getProperty ("mongodb.fortune.collection");
+
+        MongoClient mongoClient = new MongoClient (HOST, PORT);
+        MongoDatabase db = mongoClient.getDatabase (DATABASE);
+        worldCollection = db.getCollection (WORLD);
+        fortuneCollection = db.getCollection (FORTUNE);
+
+        loadData ();
```
Comment (hamiltont) [inline frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java]: @jamming please remove this call and the associated `loadData` function + `FORTUNES` data. We already setup the databases for you.

File: frameworks/Java/sabina/src/test/java/sabina/benchmark/ApplicationTest.java [hunk 3]
```diff
@@ -39,13 +38,13 @@
...
 
     private static final String ENDPOINT = "http://localhost:5050";
     private static final Gson GSON = new Gson ();
 
     @BeforeClass public static void setup () {
-        main (null);
+        Application.main (null);
     }
 
     @BeforeClass public void warm_up () throws IOException {
```
Comment (hamiltont) [inline frameworks/Java/sabina/src/test/java/sabina/benchmark/ApplicationTest.java]: We also perform an initial warmup before we actually benchmark your application, this is probably redundant :-)

**Comment by hamiltont** [comment]
looks good. Requested some minor changes, then we can merge

**Comment by jamming** [comment]
Hi, I just made the updates you proposed. Thanks for the feedback... and kudos for the great work!

**Comment by jamming** [comment]
I am not sure which is the correct place to ask this, so I'll try here (as in the issues there is no 'question' label):

In my fork's Travis build I'm getting this error:

    +timeout 15s sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10

Do you know how to fix this?

Thanks again!

**Comment by hamiltont** [comment]
> Do you know how to fix this?

Sadly, there is basically no way. You just have to keep restarting the build until that line passes. Travis-CI is known to have occasional network issues

File: frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.mongodb.*;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.bson.Document;
+
+final class MongoDbRepository implements Repository {
+    private static final String [] FORTUNES = {
```
Comment (hamiltont) [inline frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java]: Please delete this variable too

File: frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        return worlds;
+    }
+
+    private World findWorld (int id) {
+        return createWorld (worldCollection.find(eq ("_id", (double)id)).first ());
+    }
+
+    private World createWorld (Document world) {
+        try {
+            return new World (world.get ("_id", Double.class).intValue (), world.get
```
Comment (hamiltont) [inline frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java]: Just double-checking - this appears to be creating a `new World` object on every request. Is this what you intended?

Comment (jamming) [inline frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java]: This is ok... I create a World instance when a World is fetched from DB. But as fas as I see, creates an instance per findWorld call (which is the same as 'queries' supplied in the test).

**Comment by jamming** [comment]
I won't be able to make any update in the next two weeks, I would highly appreciate if this PR could make it for the 10th round... I can't wait to check the results!

**Comment by hamiltont** [comment]
lgtm, nice pr

**Pull request merged**